### PR TITLE
feat: update go-libp2p-daemon

### PR DESCRIPTION
**Pull request by MarcoPolo**
This was such a rabbit hole. I can't wait to get rid of these whole set of tools.

Challenges I ran into:

1. Where do I host these binaries so they are long lived on the IPFS network?
  1. First thought was to try web3.storage
  2. This gave me a cidv1 identifier. But this script doesn't handle cidv1. It only expects cidv0s.
  3. Fine, convert to cidv0.
  4. Mismatch in "expected SHA512". It's actually a sha256, but okay, whatever. Why?
  5. web3.storage slices the file in different size chunks than ipfs does. So when we ingest the file downloaded from the ipfs gateway we have no way of knowing what the chunks are so this will result in a different hash.
  6. Okay, I guess I can't use web3.storage.
  7. What about pinata? 
  8. That seems to work and chunk the files in a similar way. Great
2. Tests fail. 
  1. These tests rely on the tar.gz emitting a single `p2pd` file. This makes dealing with mutliple platforms hard. I changed it to accept a platform scoped binary (e.g. p2pd-darwin).
  2. The daemon now ([correctly](https://unix.stackexchange.com/questions/331611/do-progress-reports-logging-information-belong-on-stderr-or-stdout)) emits to stderr. This caused a test to fail in a confusing way.
  
  
Hopefully this unblocks @ckousik .

### CODE REVIEW

**Comment by ckousik** [comment]
Thanks @MarcoPolo ! This does unblock me, however, there seems to be an issue with running the postinstall script.
```
> go-libp2p@0.0.5 postinstall
> node src/post-install.js

Found /home/ckousik/.cache/npm-go-libp2p/QmQDjWkPejcfj8NuafB59tLubKXfCECgGTY1ayV5qjp7QR
Verifying QmQDjWkPejcfj8NuafB59tLubKXfCECgGTY1ayV5qjp7QR
OK QmQDjWkPejcfj8NuafB59tLubKXfCECgGTY1ayV5qjp7QR
Unpacked /home/ckousik/projects/libp2p/npm-go-libp2p
Linking /home/ckousik/projects/libp2p/npm-go-libp2p/bin/p2pd-linux to /home/ckousik/projects/libp2p/npm-go-libp2p/bin/p2pd
/home/ckousik/projects/libp2p/npm-go-libp2p/bin/p2pd: /home/ckousik/projects/libp2p/npm-go-libp2p/bin/p2pd: cannot execute binary file
 null
Error: Could not execute p2pd
    at link (/home/ckousik/projects/libp2p/npm-go-libp2p/src/download.js:221:11)
    at module.exports (/home/ckousik/projects/libp2p/npm-go-libp2p/src/download.js:236:10)
npm ERR! code 1
npm ERR! path /home/ckousik/projects/libp2p/npm-go-libp2p
npm ERR! command failed
npm ERR! command sh -c node src/post-install.js

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ckousik/.npm/_logs/2023-02-17T13_38_14_527Z-debug-0.log
```

**Comment by MarcoPolo** [comment]
> On inspecting the downloaded binary, it looks like the executable is compiled for ARM

It's true! I pushed a fix

**Comment by ckousik** [comment]
I was about to! Thank you!

**Comment by ckousik** [comment]
@MarcoPolo can we merge this change?

**Comment by achingbrain** [comment]
> Where do I host these binaries so they are long lived on the IPFS network?

PL used to maintain a cluster instance that you could add pins to via the #t-pinbot channel on the PL slack - that's how all previous CIDs were pinned, though I don't think the bot is running any more.

File: src/download.js
Comment (achingbrain) [inline src/download.js:30]:
```suggestion
async function cachingFetchAndVerify (url) {
```

File: src/download.js
Comment (achingbrain) [inline src/download.js:73]:
```suggestion
function unpack (url, installPath, stream) {
```

File: src/download.js
Comment (achingbrain) [inline src/download.js:101]:
```suggestion
function cleanArguments (version, platform, arch, installPath) {
```

File: src/download.js
Comment (achingbrain) [inline src/download.js:125]:
```suggestion
async function getDownloadURL (version, platform, arch, distUrl) {
```

File: src/download.js
Comment (achingbrain) [inline src/download.js:153]:
```suggestion
async function download ({ version, platform, arch, installPath, distUrl }) {
```

File: src/download.js
Comment (achingbrain) [inline src/download.js:184]:
```suggestion
async function link ({ depBin }) {
```

File: test/fixtures/clean.js
Comment (achingbrain) [inline test/fixtures/clean.js:7]:
```suggestion
module.exports = async function clean () {
```

**Review by achingbrain — COMMENTED** [review]
Just going to remove the whitespace changes as I've got some unpushed changes from 0.0.6 that need merging in and they will cause conflicts 😨

File: Makefile
Comment (achingbrain) [inline Makefile:17]: This means only amd64 is supported on linux, right?  Is there any way to have arm as well?  A bit niche but sometimes it's necessary to to run linux on a mac in a VM to debug things, which means it's running on arm.

File: src/download.js
Comment (achingbrain) [inline src/download.js:61]: Really it's b0rking because the CIDs don't match - saying it's the SHAxxx of the file is a bit misleading as it could be a different chunking algorithm that throws it off.

```suggestion
    throw new Error(`CID of ${cachedFilePath}' (${calculatedSha}) does not match expected value from ${cachedFilePath}`)
```

File: Makefile
Comment (ckousik) [reply Makefile:17]: The interop test on js-libp2p CI runs on amd64 linux. It would be easy to add a build target in the Makefile, but currently different architectures are not supported in `download.js`.

Comment (achingbrain) [reply Makefile:17]: Not going to block on this though

**Review by achingbrain — APPROVED** [review]

**Pull request merged**
